### PR TITLE
EC upgrades UI: loading state for start upgrade service & minor UI polish

### PR DESCRIPTION
**Pull request by miaawong**
#### What this PR does / why we need it:

https://app.shortcut.com/replicated/story/105489/ui-show-deployment-wizard-when-user-chooses-to-deploy-an-available-version
- loading state for start upgrade service 
- minor UI polish 
- test strict preflight check 
- test isEmbeddedCluster check
- refactor rendering of Available updates 

https://www.loom.com/share/8fecf4ecb76740c9a784a956ca66d34c

#### Which issue(s) this PR fixes:

Fixes # [sc-105489] 

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: web/src/components/apps/AppVersionHistory.tsx [hunk 1]
```diff
@@ -2083,6 +2149,30 @@ class AppVersionHistory extends Component<Props, State> {
                             )}
                           </div>
                         )}
+
+                      {/* not gitops, is embedded */}
```
Comment (miaawong) [inline web/src/components/apps/AppVersionHistory.tsx:2153]: separating the logic for rendering available updates with isEmbeddedCluster

File: web/src/components/apps/AppVersionHistory.tsx [hunk 2]
```diff
@@ -2392,16 +2482,44 @@ class AppVersionHistory extends Component<Props, State> {
...
+            >
+              <Icon
+                icon="close"
+                onClick={() =>
+                  this.setState({ shouldShowUpgradeServiceModal: false })
+                }
+                size={15}
+              />
+            </button>
+            {this.state.isStartingUpgradeService ? (
```
Comment (miaawong) [inline web/src/components/apps/AppVersionHistory.tsx:2504]: loading state

File: web/src/components/apps/AppVersionHistory.tsx [hunk 3]
```diff
@@ -975,6 +989,52 @@ class AppVersionHistory extends Component<Props, State> {
       });
   };
 
+  onCheckForUpgradeStatus = async () => {
```
Comment (sgalsaleh) [inline web/src/components/apps/AppVersionHistory.tsx:992]:
```suggestion
  onCheckForUpgradeServiceStatus = async () => {
```

File: web/src/components/apps/AppVersionHistory.tsx [hunk 4]
```diff
@@ -1435,9 +1496,12 @@ class AppVersionHistory extends Component<Props, State> {
       method: "POST",
     })
       .then(async (res) => {
+        this.state.upgradeServiceChecker.start(
```
Comment (sgalsaleh) [inline web/src/components/apps/AppVersionHistory.tsx:1499]: this should probably only be called if the request was successful

File: web/src/components/apps/AppVersionHistory.tsx [hunk 5]
```diff
@@ -1415,6 +1475,7 @@ class AppVersionHistory extends Component<Props, State> {
 
   startUpgraderService = (update: AvailableUpdate) => {
     this.setState({
+      shouldShowUpgradeServiceModal: true,
```
Comment (sgalsaleh) [inline web/src/components/apps/AppVersionHistory.tsx:1478]: we should only show the modal if the request was successful

File: web/src/components/apps/AppVersionHistory.tsx [hunk 6]
```diff
@@ -2083,6 +2149,30 @@ class AppVersionHistory extends Component<Props, State> {
                             )}
                           </div>
                         )}
+
+                      {/* not gitops, is embedded */}
+                      {!gitopsIsConnected &&
```
Comment (sgalsaleh) [inline web/src/components/apps/AppVersionHistory.tsx:2154]: you don't need to check for `gitopsIsConnected`. you can just check if it's an embedded cluster.

File: web/src/components/apps/AppVersionHistory.tsx [hunk 7]
```diff
@@ -1466,11 +1531,11 @@ class AppVersionHistory extends Component<Props, State> {
...
-      <div className="tw-max-h-[275px] tw-overflow-auto">
-        <p className="u-fontSize--normal u-fontWeight--medium tw-color-gray-800 tw-mb-2">
+      <>
+        <p className="u-fontSize--normal u-fontWeight--medium card-title tw-pb-4">
           Available Updates
         </p>
-        <div className="tw-flex tw-flex-col tw-gap-2">
+        <div className="tw-flex tw-flex-col tw-gap-2 tw-max-h-[275px] tw-overflow-auto">
           {updates.map((update, index) => (
             <div>
```
Comment (sgalsaleh) [inline web/src/components/apps/AppVersionHistory.tsx:1540]: this needs to have a `key`

**Review by sgalsaleh — APPROVED** [review]

**Comment by sgalsaleh** [comment]
I'll merge this for now to unblock ethan. Please address the comments in a separate PR.

**Pull request merged**